File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.js
function solution(left, right) {
    var answer = 0;
    
    function getFactors(num) {
        let factorsLen = 0;
        for(let i=1; i<=num; i++) {
            if(num%i === 0) {
                factorsLen++
            }
        }
        return factorsLen;
    }
    for(let j=left; j<=right; j++) {
        if(getFactors(j)%2===0) {
            answer+=j;
        } else {
            answer-=j;
        }
    }
    return answer;
}

File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.java
class Solution {
    public static int getFactors(int num) {
        int factorsLen = 0;
        for (int i = 1; i <= num; i++) {
            if (num % i == 0) {
                factorsLen++;
            }
        }
        return factorsLen;
    }
    
    public int solution(int left, int right) {
        int answer = 0;

        for (int j = left; j <= right; j++) {
            if (getFactors(j) % 2 == 0) {
                answer += j;
            } else {
                answer -= j;
            }
        }

        return answer;
    }
}